### PR TITLE
Docs: Added v3 to v4 migration guide

### PR DESCRIPTION
**Pull request by harshithkashyap**
### Description of change

v3 to v4 migration guide along with newer features section.

### CODE REVIEW

File: docs/upgrade-to-v4.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# Upgrade to V4
+
+Sequelize V4 is a major release and it introduces newer features, breaking changes and deprecations. Majority of sequelize codebase has been refactored to use ES2015 features. The following guide lists some of the changes to upgrade from v3 to v4. See [Changelog](http://docs.sequelizejs.com/manual/changelog/changelog.html#future) for full list of changes.
```
Comment (sushantdhiman) [inline docs/upgrade-to-v4.md]: 1. Changelog link can be simply replaced with `[Changelog](manual/changelog)`

2. Starting line should say `Sequelize V4 is the next major release and it introduces new features, breaking changes and deprecations`

File: docs/upgrade-to-v4.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+- Removed support for `pool:false`. To use a single connection, set `pool.max` to 1.
+- (MySQL) BIGINT now gets converted to string when number is too big
+- Removed support for referencesKey, use a references object
+    ```
+    references: {
+        key: '',
+        model: ''
+    }
+    ```
+- `classMethods` and `instanceMethods` are now deprecated.
```
Comment (sushantdhiman) [inline docs/upgrade-to-v4.md]: Not deprecated, they are removed

**Comment by sushantdhiman** [comment]
@harshithkashyap Can you run `npm run docs` on your machine and post a screenshot new page ?

**Comment by felixfbecker** [comment]
I pushed some improvements

**Pull request merged**